linux/coalescer.go

```diff
@@ -58,6 +58,12 @@ func (c *Coalescer) Coalesce(ctx context.Context, layerArtifacts []*indexer.Laye
 		for db, pkgs := range tmp {
 			dbs[db] = pkgs
 		}
+		for _, f := range artifacts.Files {
+			if c.ir.Files == nil {
+				c.ir.Files = make(map[string][]claircore.File)
+			}
```

Comment:
another option is to check if `len(artifacts.Files) > 0` before the loop and initialize there so we don't have to check `c.ir.Files == nil` each iteration